src/rust/engine/graph/src/entry.rs

```diff
@@ -594,6 +620,14 @@ impl<N: Node> Entry<N> {
     }
   }
 
+  pub fn may_have_dirty_edges(&self) -> bool {
+    match *self.state.lock() {
+      EntryState::NotStarted { was_dirty, .. } => was_dirty,
+      EntryState::Running { was_dirty, .. } => was_dirty,
```

Comment:
`NotStarted` and `Completed` definitely make sense here, but `Running` is fuzzier... I guess the idea is that when a node is dirtied while it is running, we discard the result value and move `dirty` to `was_dirty`, which would then cause us to ignore those edges? Re-reading your docstring, this makes sense. Possibly worth pointing to it in a comment here to make it more obvious that that has been considered?

Reply:
Hopefully clarified by the new wrapper enum